src/samplers/cpu/mod.rs

```diff
@@ -149,6 +120,74 @@ impl Sampler for Cpu {
 }
 
 impl Cpu {
+    #[cfg(feature = "bpf")]
+    fn bpf_enabled(&self) -> bool {
+        if self.sampler_config().bpf() {
+            for statistic in self.sampler_config().statistics() {
+                if statistic.bpf_table().is_some() {
+                    return true;
+                }
+            }
+        }
+        false
+    }
+
+    fn initialize_bpf(&mut self) -> Result<(), failure::Error> {
+        #[cfg(feature = "bpf")]
+        {
+            if self.enabled() && self.bpf_enabled() {
+                debug!("initializing bpf");
+                let code = include_str!("bpf.c");
+                let mut bpf = bcc::core::BPF::new(code)?;
+
+                for statistic in self.sampler_config().statistics() {
+                    if let Some(config) = statistic.bpf_config() {
+                        PerfEventProbe::new()
+                            .name(table.0)
+                            .event(table.1)
+                            .sample_period(SAMPLE_PERIOD)
```

Comment:
I don't think we want to specify a sample period here?

Reply:
Need exactly one of sample_period or sample_freq. 
Maybe set it to 1?

Reply:
Think we need to add error handling in the bcc crate for this too and update the docs